src/factory/deployModuleFactory.ts

```diff
@@ -21,10 +23,23 @@ export const deployModuleFactory = async (
 ): Promise<string> => {
   const singletonFactory = await getSingletonFactory(hre);
   console.log("    Singleton Factory:     ", singletonFactory.address);
-  const Factory = await hre.ethers.getContractFactory("ModuleProxyFactory");
+
+  try {
+    const Factory = await hre.ethers.getContractFactory("ModuleProxyFactory");
+    if (Factory.bytecode !== FactoryInitCode) {
+      console.warn(
+        "WARNING: The ModuleProxyFactory init code from (src/factory/contracts.ts) " +
```

Comment:
Ok nice check, good addition. One remark and one question:

1. Why is this a console.warn? It seems that this situation can only arise given a configuration misstep by a package maintainer. Shouldn't this just hard fail, throw an Error indicating the config problem?
2. I would write something slightly shorter, along the lines of: 
The compiled ModuleProxyFactory (from src/factory/contracts.ts) is outdated and does not match the bytecode stored at MasterCopyInitData[KnownContracts.FACTORY].initCod

Reply:
1. Since the check is only performed while the code is executed within the zodiac project, that thought was that it would be useful during development when working on the `ModuleProxyFactory` contract, to be able to deploy a new experimental version without having to change the hardcoded initcode. What do you think?
2. Fixed!

Reply:
Right, yes that makes sense